server/neptune/workflows/internal/activities/terraform.go

```diff
@@ -1,16 +1,75 @@
 package activities
 
-import "context"
+import (
+	"context"
+	"fmt"
+	"strings"
 
-type terraformActivities struct{}
+	"github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/terraform/helpers"
+	"github.com/runatlantis/atlantis/server/events/runtime/common"
+	"github.com/runatlantis/atlantis/server/events/terraform/ansi"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/job"
+	"github.com/uber-go/tally/v4"
+)
 
-// Terraform Init
+type terraformExec interface {
+	RunCommandAsync(ctx context.Context, jobID string, path string, args []string, customEnvVars map[string]string, v *version.Version) <-chan helpers.Line
+}
+
+type terraformActivities struct {
+	TerraformExecutor terraformExec
+	DefaultTFVersion  *version.Version
+	Scope             tally.Scope
+}
 
+// Terraform Init
 type TerraformInitRequest struct {
+	Step      job.Step
+	Envs      map[string]string
+	JobID     string
+	TfVersion string
+	Path      string
 }
 
-func (t *terraformActivities) TerraformInit(ctx context.Context, request TerraformInitRequest) error {
-	return nil
+type TerraformInitResponse struct {
+	Output string
+}
+
+func (t *terraformActivities) TerraformInit(ctx context.Context, request TerraformInitRequest) (TerraformInitResponse, error) {
+
+	version, err := version.NewVersion(request.TfVersion)
+	if err != nil {
+		return TerraformInitResponse{}, errors.Wrap(err, "resolving terraform version")
+	}
+
+	tfVersion := t.DefaultTFVersion
+	if version != nil {
+		tfVersion = version
+	}
+
+	terraformInitVerb := []string{"init"}
+	terraformInitArgs := []string{"-input=false"}
+	finalArgs := common.DeDuplicateExtraArgs(terraformInitArgs, request.Step.ExtraArgs)
+	terraformInitCmd := append(terraformInitVerb, finalArgs...)
+
+	outCh := t.TerraformExecutor.RunCommandAsync(ctx, request.JobID, request.Path, terraformInitCmd, request.Envs, tfVersion)
```

Comment:
where is this written to our log handler?

Reply:
I have a follow up PR for writing logs to the log handler. It should be similar to what we already have in async_client: 
https://github.com/lyft/atlantis/blob/65eb072e577a443a8e517656d7b0a5ff7fde1460/server/core/terraform/async_client.go#L130-L134
unless we plan to do it differently!